script.js

```diff
@@ -1,4 +1,340 @@
+var MODE_CHOOSE_GAME_MODE = `WAITING_FOR_USER_TO_CHOOSE_GAME_MODE`;
+var EITHER_HIGHEST_OR_LOWEST_WINS = `EITHER HIGHEST OR LOWEST WINS`;
+var MODE_HIGHER_NUMBER = `higher number`;
+var MODE_LOWER_NUMBER = `lower number`;
+var MODE_PLAYER_1_ROLL_DICE = `WAITING_FOR_PLAYER_1_TO_ROLL_DICE`;
+var MODE_PLAYER_1_PICK_ORDER = `WAITING_FOR_PLAYER_1_TO_PICK_DICE_ORDER`;
+var MODE_PLAYER_2_ROLL_DICE = `WAITING_FOR_PLAYER_2_TO_ROLL_DICE`;
+var MODE_PLAYER_2_PICK_ORDER = `WAITING_FOR_PLAYER_2_TO_PICK_DICE_ORDER`;
+var MODE_WINNER = `SHOW_WINNER`;
+
+//Current game mode
+var mode = MODE_CHOOSE_GAME_MODE;
+var theModeUserInput = EITHER_HIGHEST_OR_LOWEST_WINS;
+
+// Function for leaderboard for Mode: Higher number wins:
+var leaderBoardHigher = function () {
+  if (Number(P1TotalScore) > Number(P2TotalScore)) {
+    myOutputValue = `Player 1<br>Player 2`;
+  } else if (Number(P2TotalScore) > Number(P1TotalScore)) {
+    myOutputValue = `Player 2<br>Player 1`;
+  } else if (Number(P1TotalScore) == Number(P2TotalScore)) {
+    myOutputValue = `Draw`;
+  }
+  return myOutputValue;
+};
+
+// Function for leaderboard for Mode: Lower number wins:
+var leaderBoardLower = function () {
+  if (Number(P1TotalScore) < Number(P2TotalScore)) {
+    myOutputValue = `Player 1<br>Player 2`;
+  } else if (Number(P2TotalScore) < Number(P1TotalScore)) {
+    myOutputValue = `Player 2<br>Player 1`;
+  } else if (Number(P1TotalScore) == Number(P2TotalScore)) {
+    myOutputValue = `Draw`;
+  }
+  return myOutputValue;
+};
+
 var main = function (input) {
-  var myOutputValue = 'hello world';
+  var P1RandomDiceRoll1 = diceRoll();
+  var P1RandomDiceRoll2 = diceRoll();
+  var leaderBoardRanksHigher = leaderBoardHigher();
+  var leaderBoardRanksLower = leaderBoardLower();
+
+  // User to choose game mode
+  if (mode == MODE_CHOOSE_GAME_MODE) {
+    console.log(mode);
+    var myOutputValue = `Hello! Welcome to the game of BEAT THAT.<br><br>Please select your game mode preference.<br><br>Choose either "higher number" or "lower number". <br><br>Good luck!!`;
+    mode = MODE_PLAYER_1_ROLL_DICE;
+    console.log(mode);
+  } else if (input == `higher number`) {
+    theModeUserInput = MODE_HIGHER_NUMBER;
+    console.log(theModeUserInput);
+    myOutputValue = `You have selected the higher number game mode.<br><br>Player 1, please click submit to roll the dice.`;
+    mode = MODE_PLAYER_1_ROLL_DICE;
+  } else if (input == `lower number`) {
+    theModeUserInput = MODE_LOWER_NUMBER;
+    console.log(theModeUserInput);
+    myOutputValue = `You have selected the lower number game mode.<br><br>Player 1, please click submit to roll the dice.`;
+    mode = MODE_PLAYER_1_ROLL_DICE;
+  }
+
+  // Player 1's turn
+  else if (
+    mode == MODE_PLAYER_1_ROLL_DICE &&
+    (theModeUserInput == MODE_HIGHER_NUMBER ||
+      theModeUserInput == MODE_LOWER_NUMBER)
+  ) {
+    console.log(mode);
+    console.log(theModeUserInput);
+    P1DicePermutation1 = P1RandomDiceRoll1 * 10 + P1RandomDiceRoll2;
```

Comment:
Cool mathematical approach! An alternative is to treat the two numbers as a string, concaten\ate them, then cast them into a number:

```
//concat 2 strings
var concatenatedNum= "3"+ "5"

//Cast the string into a number 
var number = Number(concatenatedNum)
```
See [here](https://www.w3schools.com/jsref/jsref_number.asp) for more info on `Number()